src/models/cost_function_timeseries.jl

```diff
@@ -0,0 +1,125 @@
+function get_variable_cost(
+    ts::IS.TimeSeriesData,
+    component::Component,
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+)
+    if start_time === nothing
+        start_time = IS.get_initial_timestamp(ts)
+    end
+    data = IS.get_time_series_array(component, ts, start_time, len = len)
+    time_stamps = TimeSeries.timestamp(data)
+    return TimeSeries.TimeArray(time_stamps, map(VariableCost, TimeSeries.values(data)))
+end
+
+function get_variable_cost(
+    device::StaticInjection,
+    cost::OperationalCost;
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+)
+    time_series_key = get_variable(cost)
+    if isnothing(time_series_key)
+        error("Cost component has a `nothing` stored in field `variable`, Please use `set_variable_cost!` to add variable cost forecast.")
+    end
+    raw_data = IS.get_time_series_by_key(
+        time_series_key,
+        device;
+        start_time = start_time,
+        len = len,
+        count = 1,
+    )
+    cost = get_variable_cost(raw_data, device, start_time, len)
+    return cost
+end
+
+function get_variable_cost(
+    service::ReserveDemandCurve;
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+)
+    time_series_key = get_variable(service)
+    if isnothing(time_series_key)
+        error("Cost component has a `nothing` stored in field `variable`, Please use `set_variable_cost!` to add variable cost forecast.")
+    end
+    raw_data = IS.get_time_series_by_key(
+        time_series_key,
+        service;
+        start_time = start_time,
+        len = len,
+        count = 1,
+    )
+    cost = get_variable_cost(raw_data, service, start_time, len)
+    return cost
+end
+
+function get_services_bid(
+    device::StaticInjection,
+    cost::MarketBidCost,
+    service::Service;
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+)
+    time_series_key = get_variable(cost)
+    raw_data = IS.get_time_series_by_key(
+        time_series_key,
+        device,
+        get_name(service),
```

Comment:
Is this name different from the name inside `time_series_key`?

Reply:
yes, we decide to use the same time_series_key from variable cost field assuming that the type of forecast for service cost would be the same as  variable cost.